pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -279,67 +318,112 @@ private boolean isChannelOwner() {
+                    serviceUnit, broker, eventCounters.get(eventType).getFailure().incrementAndGet(), e);
+            throw e;
+        }
     }
 
     public CompletableFuture<Void> publishUnloadEventAsync(Unload unload) {
-        String serviceUnit = unload.serviceUnit();
-        if (isTransferCommand(unload)) {
-            ServiceUnitStateData next = new ServiceUnitStateData(Assigned,
-                    unload.destBroker().get(), unload.sourceBroker());
-            return pubAsync(serviceUnit, next).thenApply(__ -> null);
+        EventType eventType = Unload;
+        eventCounters.get(eventType).getTotal().incrementAndGet();
+        try {
+            String serviceUnit = unload.serviceUnit();
+            CompletableFuture<MessageId> future;
+            if (isTransferCommand(unload)) {
+                ServiceUnitStateData next = new ServiceUnitStateData(Assigned,
+                        unload.destBroker().get(), unload.sourceBroker());
+                future = pubAsync(serviceUnit, next);
+            } else {
+                future = tombstoneAsync(serviceUnit);
+            }
+
+            return future.whenComplete((__, ex) -> {
+                if (ex != null) {
+                    eventCounters.get(eventType).getFailure().incrementAndGet();
+                }
+            }).thenApply(__ -> null);
+        } catch (Throwable e) {
+            log.error("Failed to publish unload event. unload:{}. unloadPublishFailureCount:{}",
+                    unload, eventCounters.get(eventType).getFailure().incrementAndGet(), e);
+            throw e;
         }
-        return tombstoneAsync(serviceUnit).thenApply(__ -> null);
     }
 
     public CompletableFuture<Void> publishSplitEventAsync(Split split) {
-        String serviceUnit = split.serviceUnit();
-        ServiceUnitStateData next = new ServiceUnitStateData(Splitting, split.sourceBroker());
-        return pubAsync(serviceUnit, next).thenApply(__ -> null);
+        EventType eventType = Split;
+        eventCounters.get(eventType).getTotal().incrementAndGet();
+        try {
+            String serviceUnit = split.serviceUnit();
+            ServiceUnitStateData next = new ServiceUnitStateData(Splitting, split.sourceBroker());
+            return pubAsync(serviceUnit, next).whenComplete((__, ex) -> {
+                if (ex != null) {
+                    eventCounters.get(eventType).getFailure().incrementAndGet();
+                }
+            }).thenApply(__ -> null);
+        } catch (Throwable e) {
+            log.error("Failed to publish split event. split:{}, splitPublishFailureCount:{}",
+                    split, eventCounters.get(eventType).getFailure().incrementAndGet(), e);
+            throw e;
+        }
     }
 
     private void handle(String serviceUnit, ServiceUnitStateData data) {
+        long totalHandledRequests = getHandlerTotalCounter(data).incrementAndGet();
```

Comment:
Do we need to calculate all events for all brokers? Or record event metrics only for the target broker.

Reply:
Yes, we want to record events for all brokers.

The goal is to monitor the consistency of the tableviews across all brokers.

We need to monitor if the rate of the sum of HandlerTotalCounters( sum / 5 sec) is the same across all brokers.

Reply:
Ok, thanks for your explanation.